src/api/assets.go

```diff
@@ -3,26 +3,67 @@ package api
 import (
 	"fmt"
 	"html/template"
+	"net"
 	"net/http"
 
 	"github.com/gorilla/mux"
 )
 
-func rootHandler(w http.ResponseWriter, r *http.Request) {
-	data, err := Asset("static/index.htm")
-	if err != nil {
-		fmt.Println(err)
-		w.WriteHeader(404)
-		return
+type bindInfo struct {
+	ApiUrl string
+	IP4    string
+	IP6    string
+	Port   string
+}
+
+type templateData struct {
+	ApiUrl    string
+	Primary   string
+	Secondary string
+}
+
+func (info *bindInfo) templateFor(r *http.Request) *templateData {
+	data := &templateData{
+		ApiUrl: info.ApiUrl,
 	}
-	tmpl, err := template.New("root").Parse(string(data))
-	if err != nil {
-		fmt.Println(err)
-		w.WriteHeader(500)
-		return
+	addr, _, _ := net.SplitHostPort(r.RemoteAddr)
+	if len(net.ParseIP(addr)) == net.IPv6len {
+		// client is ipv6
+		data.Primary = info.IP6
+		data.Secondary = info.IP4
+	} else {
+		// client is ipv4
+		data.Primary = info.IP4
+		data.Secondary = info.IP6
+	}
+	// If the user didn't provide one of the two addresses, we pass the UI an
+	// empty string.
+	if data.Primary != "" {
+		data.Primary = net.JoinHostPort(data.Primary, info.Port)
+	}
+	if data.Secondary != "" {
+		data.Secondary = net.JoinHostPort(data.Secondary, info.Port)
+	}
```

Comment:
We can improve this bit but I am happy to do it in a separate diff so I get to mess around this part of the code.
Basically, if the connections is HTTP and port is 80, we dont need to happen the port number.
if the connections is HTTPS and port is 443, same story.
Also, at this stage, we assume we talk plain HTTP because we don't provide a hint of HTTP or HTTPS. Again, I am happy to deal with it in a separate diff.